### PR TITLE
MOB-20950 BasicPushTemplate, AEPPushTemplate tests

### PR DESCRIPTION
**Pull request by navratan-soni**
Added Test Cases for BasicPushTemplate, AEPPushTemplate, added required data in MockDataProvider

## Description

This PR includes the addition of test cases for the `BasicPushTemplate` class. The tests cover initialization with all keys present, initialization with invalid JSON, and initialization with malformed JSON.
Also, basic sanity and Illeagle argument cases in `AEPPushTemplate` class

## Related Issue
https://jira.corp.adobe.com/browse/MOB-20950

## Motivation and Context

Added test cases from scratch for these classes, which cases did not exist before

## How Has This Been Tested?

Written Unit Test Cases

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/AEPPushTemplateTest.kt [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        assertEquals(MOCKED_BODY, aepPushTemplate.body)
+        assertEquals(MOCKED_PAYLOAD_VERSION, aepPushTemplate.payloadVersion)
+    }
+
+    @Test
+    fun `Test exception with missing data adb_title`() {
+        val aepPushBasicData = MockAEPPushTemplateDataProvider.getMockedDataMapWithRequiredData()
+        aepPushBasicData.remove(PushTemplateConstants.PushPayloadKeys.TITLE)
+        val exception = assertFailsWith<IllegalArgumentException> {
+            BasicPushTemplate(MapData(aepPushBasicData))
```
Comment (spoorthipujariadobe) [inline code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/AEPPushTemplateTest.kt:42]: Name of the file says its tests for `AEPPushTemplate` but we're creating `BasicPushTemplate` here. Is it supposed to be moved to a different file or is the class under test wrong?

File: code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/AEPPushTemplateTest.kt [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+    @Test
+    fun `Test AEPPushTemplate initialization with all data`() {
+        val aepPushTemplate = provideMockedBasicPushTemplateWithAllKeys()
+        assertEquals(PushTemplateType.BASIC, aepPushTemplate.templateType)
+        assertEquals(MOCKED_TAG, aepPushTemplate.tag)
+        assertFalse(aepPushTemplate.isFromIntent)
+        assertEquals(MOCKED_ACTION_URI, aepPushTemplate.actionUri)
+        assertEquals(PushTemplateConstants.ActionType.NONE, aepPushTemplate.actionType)
+        assertEquals(2, aepPushTemplate.actionButtonsList?.size)
```
Comment (spoorthipujariadobe) [inline code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/AEPPushTemplateTest.kt:84]: `actionButtonsList` is a part of `BasicPushTemplate`, not `AEPPushTemplate`

File: code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/AEPPushTemplateTest.kt [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        assertEquals(MOCKED_PRIORITY, aepPushTemplate.getNotificationPriority())
+        assertEquals(
+            NotificationVisibility.getNotificationCompatVisibilityFromString(
+                MOCKED_VISIBILITY
+            ),
+            aepPushTemplate.getNotificationVisibility()
+        )
+        assertEquals("PRIORITY_DEFAULT", aepPushTemplate.priorityString)
+        assertEquals("PUBLIC", aepPushTemplate.visibilityString)
+        assertEquals("remind me", aepPushTemplate.remindLaterText)
```
Comment (spoorthipujariadobe) [inline code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/AEPPushTemplateTest.kt:102]: Remind later is part of BasicPushTemplate, please move this to `BasicPushTemplateTests.kt`

File: code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/AEPPushTemplateTest.kt [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+        assertEquals("1234567890", aepPushTemplate.remindLaterTimestamp.toString())
```
Comment (spoorthipujariadobe) [inline code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/AEPPushTemplateTest.kt:103]: Same comment here, please move this to BasicPushTemplateTests.kt

File: code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/AEPPushTemplateTest.kt [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+            aepPushTemplate.getNotificationImportance(),
+            NotificationManager.IMPORTANCE_DEFAULT
+        )
+    }
+
+    @Test
+    fun `Test AEPPushTemplate initialization with null priority`() {
+        val data = MockAEPPushTemplateDataProvider.getMockedAEPDataMapWithAllKeys()
+        data.remove(PushTemplateConstants.PushPayloadKeys.PRIORITY)
+        val aepPushTemplate = BasicPushTemplate(MapData(data))
```
Comment (spoorthipujariadobe) [inline code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/AEPPushTemplateTest.kt:121]: tests are for `AEPPushTemplate` but we're creating `BasicPushTemplate` here

File: code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/AEPPushTemplateTest.kt (hunk 1 above)
Comment (rymorale) [reply code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/AEPPushTemplateTest.kt:42]: i think its using a basic push template as an AEPPushTemplate can't be created directly as it's sealed. i think this is ok as the verification is checking for AEPPushTemplate KVP.

Comment (spoorthipujariadobe) [reply code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/AEPPushTemplateTest.kt:42]: Ah ok, that makes sense. Thank you!

**Review by rymorale — APPROVED** [review]
i think the tests look good. Feel free to merge after addressing Spoorthi's comments.

File: code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/AEPPushTemplateTest.kt (hunk 3 above)
Comment (navratan-soni) [reply code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/AEPPushTemplateTest.kt]: Done removed, these assertions exist in BasicPushTemplate.

**Comment by navratan-soni** [comment]
Done the suggested changes.

**Pull request merged**